_pytest/capture.py

```diff
@@ -308,6 +308,14 @@ class FDCapture:
     """ Capture IO to/from a given os-level filedescriptor. """
 
     def __init__(self, targetfd, tmpfile=None):
+        # ensure readline is imported so that it attaches to the correct
+        # stdio handles on Windows
+        if targetfd in (0, 1, 2):
```

Comment:
A small suggestion which I think will improve the code a little: move this block to a function (not method) `readline_workaround` or something similar. 

This way it doesn't get mixed with the code around it and you can put a detailed explanation in its docstring. Also, include a link to the relevant issue in the `readline` repo that you mention you are working on, as well as a link to this PR so it links to your excellent explanation to the problem.